internal/pkg/daemon/seccompprofile/seccompprofile.go

```diff
@@ -374,6 +349,48 @@ func (r *Reconciler) reconcileSeccompProfile(
 	return reconcile.Result{}, nil
 }
 
+func (r *Reconciler) reconcileDeletingSeccompProfile(
+	ctx context.Context, nodeStatus *nodestatus.StatusClient,
+	sp *v1alpha1.SeccompProfile, l logr.Logger,
+) (reconcile.Result, error) {
+	// object is being deleted
+	setTerminatingStatus := func(sp *v1alpha1.SeccompProfile) *v1alpha1.SeccompProfileStatus {
+		status := sp.Status.DeepCopy()
+		status.SetConditions(rcommonv1.Deleting())
+		status.Status = secprofnodestatusv1alpha1.ProfileStateTerminating
+		return status
+	}
+
+	if err := r.setBothStatuses(ctx, nodeStatus, sp, setTerminatingStatus); err != nil {
+		l.Error(err, "cannot update SeccompProfile status")
+		r.record.Event(sp, event.Warning(reasonCannotUpdateProfile, err))
+		return reconcile.Result{}, errors.Wrap(err, "updating status for deleted SeccompProfile")
+	}
+
+	if controllerutil.ContainsFinalizer(sp, hasActivePodsFinalizerString) {
+		l.Info("Profile still in use, requeue")
```

Comment:
nit: could mention in the message that this is a deletion event, e.g. `"cannot delete profile in use by pod, requeuing"`

Reply:
Done. Sorry it took so long to get back to your review; I was away for several weeks.